File: Algorithms/Array/Kadane's Algorithm/Maximum_sum_subarray/maximum_sum_subarray.cpp
#include <iostream>
using namespace std;


int max_sum_subarray(int array,int n){
    int max_sum = INT_MIN;
    int cur_max = 0;

    for(int i = 0; i < n; i++){
        cur_max = cur_max + array[i];

        if( cur_max > max_sum){
            max_sum = cur_max;
        }
        if(cur_max < 0){
            cur_max = 0;
        }
    }
}

int main(){
    int arr_1[] = {5,6,-12,10,2,-1};
    int arr_2[] = {-5,-6,-7,-1,0};

    cout<<max_sum_subarray(arr_1,6);
    cout<<max_sum_subarray(arr_2,5);

    return 0; 
}

File: Algorithms/Array/Kadane's Algorithm/Maximum_sum_subarray/maximum_sum_subarray.py
from sys import maxsize

def max_subarray_sum(array):
    max_sum = - maxsize - 1
    cur_sum = 0
    
    for num in array:
        cur_sum += num
        if cur_sum > max_sum:
            max_sum = cur_sum
        if cur_sum < 0:
            cur_sum = 0
    return max_sum
    

#driver code for python file
if __name__=="__main__":
    arr_1 = [5,6,-12,12,2,-1]
    arr_2 = [-2,-3,-8,-5,-20,-15]
    print("for array ",arr_1,"\nMax sum of a subarray is ", max_subarray_sum(arr_1))
    print("for array ",arr_2,"\nMax sum of a subarray is ", max_subarray_sum(arr_2))